### DISCUSSION TITLE
Handling Pairing popup dialog

### DISCUSSION THREAD

**Discussion by pedropastor** [PhilipsHue/flutter_reactive_ble#493] [Q&A]
I am having problems handling the pairing dialog that pops up whenever you try to connect a device that is adevertising itself.

Also, interacting in any way with the dialog disconnects the device, and needs reconnection, which will pop up the dialog again.

I am working with BLE devices that do not need pairing, and in some cases, due to the business case, CANNOT be paired.

Any idea how to override the pairing behavior of Android/iOS? worse case, how can I detect the pairing already exists or not to expect that disconnection until paired?

I have been completely stuck in this for several days

**Comment by remonh87** [comment]
@pedropastor  for some reason we missed this issue hope it is solved.  I recommend when scanning to use the following filter: 
https://github.com/PhilipsHue/flutter_reactive_ble/blob/8d66469df06e814a96eed48174d34525459c62c5/packages/flutter_reactive_ble/lib/src/reactive_ble.dart#L228
 
to filter out devices that do not belong to you.

**Reply by paulVulog** [reply]
I guess I was facing the same issue recently on Android.
In fact, you probably already tried to manually pair with your bluetooth device (using the android system - not your app)
If so, then you will always receive a Pairing Popup on every .connect() from your app. 
(in my case, I get a Pairing popup with a cancel/ok button, but whatever you click, the popup disapears and you have like 2 seconds until you get a disconnection.
If I communicate within this time laps, then I am able to keep the connection alive.

---

Solution : 
I was required to Reset all the device network settings. If you don't have this option, then you need to Factory reset... :(
